src/frequencypage.cpp

```diff
@@ -0,0 +1,149 @@
+/*
+  frequencypage.cpp
+
+  This file is part of Hotspot, the Qt GUI for performance analysis.
+
+  Copyright (C) 2021 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+  Author: Lieven Hey <lieven.hey@kdab.com>
+
+  Licensees holding valid commercial KDAB Hotspot licenses may use this file in
+  accordance with Hotspot Commercial License Agreement provided with the Software.
+
+  Contact info@kdab.com if any conditions of this licensing are not clear to you.
+
+  This program is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 2 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "frequencypage.h"
+
+#include <KChart/KChart>
+#include <KChart/KChartCartesianAxis>
+#include <KChart/KChartPlotter>
+#include <KColorScheme>
+
+#include "models/frequencymodel.h"
+#include "parsers/perf/perfparser.h"
+#include "util.h"
+#include "filterandzoomstack.h"
+
+namespace {
+class TimeAxis : public KChart::CartesianAxis
+{
+    Q_OBJECT
+public:
+    explicit TimeAxis(KChart::AbstractCartesianDiagram* diagram = nullptr)
+        : CartesianAxis(diagram)
+    {
+    }
+
+    const QString customizedLabel(const QString& label) const override
+    {
+        // TODO: substract start time to display useful timestamps
```

Comment:
to fix this:

in `Data::Summary`, replace the `quint64 applicationRunningTime` with the `Data::TimeRange applicationTime;` from the `perfparser.cpp`. then you access the start from there

Reply:
this TODO can now be removed